.github/workflows/enos-run.yml

```
@@ -75,43 +77,40 @@ jobs:
           mkdir -p ./enos/support
           echo "${{ secrets.ENOS_CI_SSH_KEY }}" > ./enos/support/private_key.pem
           chmod 600 ./enos/support/private_key.pem
-      - name: Set up Bats CLI UI tests dependency cache
-        if: matrix.test == 'cli_ui'
+      - name: Set up dependency cache
         id: dep-cache
         uses: actions/cache@v3
         with:
           path: /tmp/bats-cli-ui-deps
           key: enos-bats-cli-ui-deps-jq-1.6-password-store-1.7.4
       - name: Set up Node for Bats install
-        if: matrix.test == 'cli_ui'
+        if: matrix.filter == 'integration test:cli_ui'
         uses: actions/setup-node@v3
         with:
           node-version: 16
       - name: Install Bats via NPM
-        if: matrix.test == 'cli_ui'
         # Use npm so this workflow is portable on multiple runner distros
         run: npm install --location=global bats
       - name: Download jq for Bats CLI UI tests
-        if: matrix.test == 'cli_ui' && steps.dep-cache.outputs.cache-hit != 'true'
+        if: matrix.filter == 'integration test:cli_ui' && steps.dep-cache.outputs.cache-hit != 'true'
         # NOTE: if you update the jq version make sure to update the dep cache key
         run: |
           mkdir -p /tmp/bats-cli-ui-deps
           wget https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 -O /tmp/bats-cli-ui-deps/jq-bin
       - name: Install jq for Bats CLI UI tests
-        if: matrix.test == 'cli_ui'
+        if: matrix.filter == 'integration test:cli_ui' && steps.dep-cache.outputs.cache-hit != 'true'
         run: |
           chmod +x /tmp/bats-cli-ui-deps/jq-bin
           sudo cp /tmp/bats-cli-ui-deps/jq-bin /usr/local/bin/jq
```

In one of the new scenarios I'm adding, it will also need `jq`. I presume we'll need to add another block here for that?

---

You make a good point implicitly that this could grow to be unwieldy. I think one approach would be for us to make the inversion of this logic instead, but maybe we could more simply _always_ install `jq` as it won't harm scenarios that _don't_ use it.

---

```suggestion
        run: |
          chmod +x /tmp/bats-cli-ui-deps/jq-bin
          sudo cp /tmp/bats-cli-ui-deps/jq-bin /usr/local/bin/jq
```

What do we think about removing it?

---

Seems ok? Reminder: we'd have to remove it in the download step above, too.

Alternatively, would it make sense for me to handle this in the scenario? Would that even be possible?

---

@jaymalasinha do you think that ensuring `jq` is present should be handled in the GHA workflow or Enos scenario?

---

`jq` install can be done within the Enos scenario modules

---

Hm but do you think we should remove it here and edit all of the scenarios that need it to include it...

or should we delegate that task to the workflow?

I guess my question is that we don't really want to be doing this in different ways that get stuffed into the same workflow (some installing jq with Enos, some doing it with the GHA workflow).

---

I talked with Michael about this out of the comments- we're going to install it as part of the workflow as we already assume other local deps (like Go) are installed when this is running on someone's MacBook.